crates/dids/src/method/dht/mod.rs

```diff
@@ -0,0 +1,148 @@
+mod bep44;
+mod convert;
+mod registered_types;
+
+use std::sync::Arc;
+
+use crypto::Curve;
+use keys::key_manager::KeyManager;
+use reqwest::blocking::Client;
+
+use crate::{
+    bearer::BearerDid,
+    document::{Document, Service, VerificationMethod},
+    identifier::Identifier,
+    resolver::{ResolutionMetadata, ResolutionResult},
+};
+
+use self::{
+    bep44::encode_bep44_message, convert::document_packet::document_to_packet,
+    registered_types::RegisteredDidType,
+};
+
+use super::{Method, MethodError};
+
+const DEFAULT_TTL: u32 = 7200; // seconds
+const DEFAULT_GATEWAY_URL: &str = "https://diddht.tbddev.org";
+const DID_DHT_SPEC_VERSION: u32 = 0;
+const NAME: &'static str = "dht";
+
+/// Concrete implementation for a did:dht DID
+pub struct DidDht;
+
+/// Options that can be used to create a did:dht DID
+pub struct DidDhtCreateOptions {
+    pub publish: bool,
+    // TODO: gatewayUri - default 'https://diddht.tbddev.org'
+    pub also_known_as: Option<Vec<String>>,
+    pub controller: Option<Vec<String>>,
+    pub service: Option<Vec<Service>>,
+    pub registered_type: Option<Vec<RegisteredDidType>>,
+    pub verification_methods: Option<Vec<VerificationMethod>>,
+}
+
+impl Method<DidDhtCreateOptions> for DidDht {
+    const NAME: &'static str = "dht";
+
+    fn create(
+        key_manager: Arc<dyn KeyManager>,
+        options: DidDhtCreateOptions,
+    ) -> Result<BearerDid, MethodError> {
+        // Generate private and public keypair
+        let key_alias = key_manager.generate_private_key(
+            Curve::Ed25519,
+            Some("0".to_string()), // PR review: i copied this from jwk.rs. do we want this to be None?
+        )?;
+        let public_key = key_manager.get_public_key(&key_alias)?;
+
+        // Create did uri
+        let public_jwk = public_key.jwk()?;
+        let jwk_string = serde_json::to_string(public_jwk.as_ref()).map_err(|_| {
+            MethodError::DidCreationFailure("failed to serialize public jwk".to_string())
+        })?;
+        let identifier = zbase32::encode_full_bytes(jwk_string.as_bytes());
```

Comment:
as far as I can tell the spec would call this either a suffix or identity_key

```suggestion
        let suffix = zbase32::encode_full_bytes(jwk_string.as_bytes());
```

or 

```suggestion
        let identity_key = zbase32::encode_full_bytes(jwk_string.as_bytes());
```

Reply:
fyi the [Terminology in the spec](https://did-dht.com/#terminology) was ambiguous on this matter but it's been iron'd out -- the proper variable name here is `suffix`